grpc/generate_endpointspec_test.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package grpc_test
+
+import (
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	. "github.com/DataDog/chaos-controller/grpc"
+	pb "github.com/DataDog/chaos-controller/grpc/disruption_listener"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("construct DisruptionListener query for configuring disruptions from api spec", func() {
+	var (
+		endpointAlterations []chaosv1beta1.EndpointAlteration
+		endpointSpec        []*pb.EndpointSpec
+	)
+
+	Context("with five alterations which add up to less than 100", func() {
+		BeforeEach(func() {
+			endpointAlterations = []chaosv1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "service/api_1",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     25,
+				},
+				{
+					TargetEndpoint:   "service/api_2",
+					ErrorToReturn:    "PERMISSION_DENIED",
+					OverrideToReturn: "",
+					QueryPercent:     50,
+				},
+				{
+					TargetEndpoint:   "service/api_1",
+					ErrorToReturn:    "ALREADY_EXISTS",
+					OverrideToReturn: "",
+					QueryPercent:     20,
+				},
+				{
+					TargetEndpoint:   "service/api_2",
+					ErrorToReturn:    "NOT_FOUND",
+					OverrideToReturn: "",
+					QueryPercent:     0,
+				},
+				{
+					TargetEndpoint:   "service/api_1",
+					ErrorToReturn:    "",
+					OverrideToReturn: "{}",
+					QueryPercent:     0,
+				},
+			}
+
+			var err error
+			endpointSpec = GenerateEndpointSpecs(endpointAlterations)
+			Expect(err).To(BeNil())
+		})
+
+		It("should create a list of endpointSpecs with 2 elements", func() {
+			Expect(len(endpointSpec)).To(Equal(2))
+		})
+
+		It("should create and endpointSpec for api_1 with 3 elements", func() {
+			var endpointSpec_1 *pb.EndpointSpec
+			if endpointSpec[0].TargetEndpoint == "service/api_1" {
```

Comment:
Is this because the order is indeterminate? I think that's worth commenting here for future readers